gobblin-data-management/src/test/java/org/apache/gobblin/data/management/dataset/ManifestBasedDatasetFinderTest.java

```diff
@@ -44,4 +59,36 @@ public void testFindDataset()
     List<ManifestBasedDataset> datasets = finder.findDatasets();
     Assert.assertEquals(datasets.size(), 1);
   }
-}
+
+  @Test
+  public void testFindFiles() throws IOException, URISyntaxException {
+
+    //Get manifest Path
+    Path manifestPath = new Path(getClass().getClassLoader().getResource("manifestBasedDistcpTest/sampleManifest.json").getPath());
+
+    // Test manifestDatasetFinder
+    Properties props = new Properties();
+    props.setProperty(ConfigurationKeys.DATA_PUBLISHER_FINAL_DIR, "/");
+
+    try (FileSystem destFs = Mockito.mock(FileSystem.class); FileSystem sourceFs = Mockito.mock(FileSystem.class)) {
+      URI SRC_FS_URI = new URI("source", "the.source.org", "/", null);
+      URI DEST_FS_URI = new URI("dest", "the.dest.org", "/", null);
+      Mockito.when(sourceFs.getUri()).thenReturn(SRC_FS_URI);
+      Mockito.when(destFs.getUri()).thenReturn(DEST_FS_URI);
+      Mockito.when(sourceFs.getFileStatus(any(Path.class))).thenReturn(localFs.getFileStatus(manifestPath));
+      Mockito.when(sourceFs.exists(any(Path.class))).thenReturn(true);
+      Mockito.when(sourceFs.open(manifestPath)).thenReturn(localFs.open(manifestPath));
+      Mockito.when(destFs.exists(any(Path.class))).thenReturn(false);
+      Mockito.doAnswer(invocation -> {
+        Object[] args = invocation.getArguments();
+        Path path = (Path)args[0];
+        return localFs.makeQualified(path);
+      }).when(sourceFs).makeQualified(any(Path.class));
+      Iterator<FileSet<CopyEntity>> fileSets =
+          new ManifestBasedDataset(sourceFs, manifestPath, props).getFileSetIterator(destFs, CopyConfiguration.builder(destFs, props).build());
+      Assert.assertTrue(fileSets.hasNext());
+      FileSet<CopyEntity> fileSet = fileSets.next();
+      Assert.assertEquals(fileSet.getFiles().size(), 2);
```

Comment:
Yep, I just want to make sure parser works for the manifest and we are able to get the correct number of file. But the copy entity is pretty corrupted as I give the mocked file status here.
It will make more sense for us to test copy entity if I'm testing writing manifest.